src/main/java/focusedCrawler/crawler/async/HttpDownloader.java

```diff
@@ -8,18 +8,16 @@
 import java.io.PrintWriter;
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.net.UnknownHostException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Future;
-import java.util.concurrent.LinkedBlockingQueue;
-import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.ThreadPoolExecutor;
-import java.util.concurrent.TimeUnit;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.*;
```

Comment:
Apparently there are no relevant changes in this file for the new feature, so you can leave these changes out to make the PR smaller, easier to review, and test. You could make formatting changes in a separate commit/PR as well if needed.

There are other files with small formatting changes like this.

We try to follow the Google Style Guide: https://github.com/google/styleguide (but using 4 spaces for tabs instead of 2)
If you haven't done it yet, you can install the configuration file in your IDE. As per Google Style Guide, wildcard imports should be avoided, but if import the configurations, your IDE will do the right thing automatically.